studyBuddy/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,133 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson; 
+
+import com.google.sps.models.Student;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    // Value to filter students by while Querying.
+    private String subject;
+
+/********************
+Handles GET requests
+********************/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // TODO Integrate UserService
+    final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    // Query Datastore
+    Filter subject_filter = new FilterPredicate("subject", FilterOperator.EQUAL, this.subject);
```

Comment:
I believe this is only filtering by subject, which will work. But I don't know if we also wanted to make sure that the schools matched too.

Reply:
Yes. We would want to filter by School as well. I will add that once I integrate the user authentication feature because that is where we will grab information about the school.
For now, it is just filtering based on the information coming from the search page ie **subject**

Thanks.

Reply:
Sounds good.